server/src/main/java/org/apache/iotdb/db/service/basic/ServiceProvider.java

```diff
@@ -61,40 +59,55 @@
 
 import static org.apache.iotdb.db.utils.ErrorHandlingUtils.onNPEOrUnexpectedException;
 
-public class BasicServiceProvider {
+public abstract class ServiceProvider {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(BasicServiceProvider.class);
-  protected static final Logger AUDIT_LOGGER =
+  private static final Logger LOGGER = LoggerFactory.getLogger(ServiceProvider.class);
+  public static final Logger AUDIT_LOGGER =
       LoggerFactory.getLogger(IoTDBConstant.AUDIT_LOGGER_NAME);
-  protected static final Logger SLOW_SQL_LOGGER =
+  public static final Logger SLOW_SQL_LOGGER =
       LoggerFactory.getLogger(IoTDBConstant.SLOW_SQL_LOGGER_NAME);
 
-  protected static final TSProtocolVersion CURRENT_RPC_VERSION =
+  public static final TSProtocolVersion CURRENT_RPC_VERSION =
       TSProtocolVersion.IOTDB_SERVICE_PROTOCOL_V3;
 
-  protected static final IoTDBConfig CONFIG = IoTDBDescriptor.getInstance().getConfig();
+  public static final IoTDBConfig CONFIG = IoTDBDescriptor.getInstance().getConfig();
+
+  public static final QueryTimeManager QUERY_TIME_MANAGER = QueryTimeManager.getInstance();
+  public static final TracingManager TRACING_MANAGER = TracingManager.getInstance();
+  public static final QueryFrequencyRecorder QUERY_FREQUENCY_RECORDER =
+      new QueryFrequencyRecorder(CONFIG);
 
-  protected final QueryTimeManager queryTimeManager = QueryTimeManager.getInstance();
-  public static SessionManager sessionManager = SessionManager.getInstance();
-  protected final TracingManager tracingManager = TracingManager.getInstance();
-  protected final QueryFrequencyRecorder queryFrequencyRecorder;
+  public static SessionManager SESSION_MANAGER = SessionManager.getInstance();
 
-  protected Planner processor;
-  protected IPlanExecutor executor;
+  private final Planner processor;
```

Comment:
why not use planner?

Reply:
I found that the original code was processor, so I didn't modify it. But I think planner is better. I've changed it to planner.